src/Simplex/Messaging/Agent/Store/SQLite/Schema.hs

```diff
@@ -77,27 +77,59 @@ connections =
       snd_host TEXT,
       snd_port TEXT,
       snd_id BLOB,
-      PRIMARY KEY(conn_alias),
-      FOREIGN KEY(rcv_host, rcv_port, rcv_id) REFERENCES rcv_queues(host, port, rcv_id),
-      FOREIGN KEY(snd_host, snd_port, snd_id) REFERENCES snd_queues(host, port, snd_id)
+      PRIMARY KEY (conn_alias),
+      FOREIGN KEY (rcv_host, rcv_port, rcv_id) REFERENCES rcv_queues (host, port, rcv_id),
+      FOREIGN KEY (snd_host, snd_port, snd_id) REFERENCES snd_queues (host, port, snd_id)
     ) WITHOUT ROWID;
   |]
 
 messages :: Query
 messages =
   [sql|
     CREATE TABLE IF NOT EXISTS messages(
-      agent_msg_id INTEGER NOT NULL,
+      agent_msg_id INTEGER PRIMARY KEY,
+      agent_timestamp TEXT NOT NULL,
       conn_alias TEXT NOT NULL,
-      timestamp TEXT NOT NULL,
+      rcv_msg_id INTEGER,
+      snd_msg_id INTEGER,
       message BLOB NOT NULL,
-      direction TEXT NOT NULL,
       msg_status TEXT NOT NULL,
-      PRIMARY KEY(agent_msg_id, conn_alias),
-      FOREIGN KEY(conn_alias) REFERENCES connections(conn_alias)
-    ) WITHOUT ROWID;
+      FOREIGN KEY (conn_alias) REFERENCES connections (conn_alias),
+      FOREIGN KEY (rcv_msg_id) REFERENCES rcv_messages (rcv_msg_id),
+      FOREIGN KEY (snd_msg_id) REFERENCES snd_messages (snd_msg_id),
```

Comment:
wouldn't it be better to have these IDs in queues? Particularly for non-duplex connections - they seem to belong there...

Reply:
ah - sorry - got confused... this is still separate tables design...